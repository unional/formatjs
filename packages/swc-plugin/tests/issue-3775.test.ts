import {transform} from '@swc/core'
import FormatJSTransformer from '..'

const input = `
const msgs = defineMessages({
  message: {
    defaultMessage: \`Hello World!\`,
    description: 'The default message',
  },
});
`

test('GH issue #3775', async function () {
  const out = await transform(input, {
    filename: 'test.ts',
    jsc: {
      parser: {
        syntax: 'typescript',
        decorators: true,
        dynamicImport: true,
      },
    },
    plugin: m =>
      new FormatJSTransformer({
        filename: 'test.js',
        overrideIdFn: '[hash:base64:10]',
      }).visitProgram(m),
  })
  expect(out).toMatchSnapshot()
})
