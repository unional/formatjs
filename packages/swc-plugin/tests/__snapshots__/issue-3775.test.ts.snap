// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GH issue #3775 1`] = `
Object {
  "code": "var msgs = defineMessages({
    message: {
        id: \\"GCdcoTfS3U\\",
        defaultMessage: \\"Hello World!\\"
    }
});
",
}
`;
